middleware/src/authentication/authentication.go

```diff
@@ -0,0 +1,95 @@
+package authentication
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"log"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+type Claims struct {
+	Username string `json:"username"`
+	jwt.StandardClaims
+}
+
+type JwtAuth struct {
+	jwtKey string
+}
+
+func NewJwtAuth() *JwtAuth {
+	jwtAuth := &JwtAuth{}
+	// generate random string with 32 bytes of entropy and use it as the signing key
+	JwtKey, err := jwtAuth.generateRandomString(32)
+	if err != nil {
+		log.Println("could not get enough entropy to generate key")
```

Comment:
>```go 
>// GenerateRandomBytes returns securely generated random bytes. It will return an error
>// if the system's secure random number generator fails to function correctly, in which
>// case the caller should not continue.
>```

Did I miss something or do you just continue if there is not enough entropy?

Reply:
If you check the entire NewMiddleware method, I am not returning from the setup if any error occurs. This is obviously bad. I think we should panic if these setup functions fail, what do you think? Alternatively we could also return the NewMiddleware method with an error and then exit the main process gracefully. I think in this place I prefer a panic though.

Reply:
I discussed with @Stadicus offline that I'll just log the error and exit with a fatal log message in the main function if there is a redis problem at the middleware startup. 
In the case of missing entropy or another problem with the jwt start I'll let the NewMiddleware function panic, since this should never happen on a device like ours.

Reply:
Please don't panic. Panic is for a programmer's (human) error. Lack of entropy is not a human error.

Reply:
Best is to return the error and let the caller handle it.

Reply:
Thanks Alex, dropped the panic and returned the error instead.